typescript/src/httpHandlerFactory.ts

```diff
@@ -0,0 +1,47 @@
+import { HttpHandler, IHttpHandler } from "./httpHandler";
+
+function getPriorityBaseUrl(): string {
+  if (process.env.ORCHESTRATE_BASE_URL) {
+    return process.env.ORCHESTRATE_BASE_URL;
+  }
+  return "https://api.careevolutionapi.com";
+}
+
+function getPriorityApiKey(apiKey: string | undefined): string | undefined {
+  if (apiKey) {
+    return apiKey;
+  }
+  if (process.env.ORCHESTRATE_API_KEY) {
+    return process.env.ORCHESTRATE_API_KEY;
+  }
+  return undefined;
+}
+
+function getAdditionalHeaders(): { [key: string]: string; } | undefined {
+  if (process.env.ORCHESTRATE_ADDITIONAL_HEADERS) {
+    return JSON.parse(process.env.ORCHESTRATE_ADDITIONAL_HEADERS);
+  }
+  return undefined;
+}
+
+export function createHttpHandler(
+  apiKey: string | undefined,
+): IHttpHandler {
+  const baseUrl = getPriorityBaseUrl();
+  const additionalHeaders = getAdditionalHeaders();
+
+  const defaultHeaders = {
+    ...(additionalHeaders ?? {}),
+    ...{
```

Comment:
I don't think you need the spread operator here. 

Also would prefer declaring the type of the object over casting. You get a little more type safety that way

```
  const defaultHeaders: { [key: string]: string } = {
      "Content-Type": "application/json",
      Accept: "application/json",
      ...additionalHeaders,
    
  }
```